junit-platform-engine/src/main/java/org/junit/platform/engine/EngineDiscoveryIssue.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.engine;
+
+import java.util.Optional;
+
+public interface EngineDiscoveryIssue {
+
+	static EngineDiscoveryIssue create(Severity severity, String message) {
+		return builder(severity, message).build();
+	}
+
+	static Builder builder(Severity severity, String message) {
+		return new DefaultEngineDiscoveryIssue.Builder(severity, message);
+	}
+
+	Severity severity();
+
+	String message();
+
+	Optional<DiscoverySelector> selector();
+
+	Optional<TestSource> source();
+
+	Optional<Throwable> cause();
+
+	enum Severity {
+
+		/**
+		 * Indicates that the engine encountered something that could
+		 * be potentially problematic, but could also happen due to a valid
+		 * setup or configuration.
+		 */
+		NOTICE,
+
+		/**
+		 * Indicates that a deprecated feature was used that might be
+		 * removed or change behavior in a future release.
+		 */
+		DEPRECATION,
+
+		/**
+		 * Indicates that the engine encountered something that is
+		 * problematic and might lead to unexpected behavior.
+		 */
+		WARNING
+	}
```

Comment:
:speech_balloon: These are "inspired" by the [issue severities that PHPUnit uses](https://docs.phpunit.de/en/10.5/error-handling.html).

Reply:
Why is there no `ERROR` severity?

Reply:
My initial thinking was that errors would always be "reported" by throwing exceptions. However, I now tend to think that we should include them here so the handling is consistent and all issues are reported rather than just the first error.